travis/setup_dependencies_common.sh

```diff
@@ -221,23 +225,29 @@ fi
 # NUMPY
 # We use --no-pin to avoid installing other dependencies just yet.
 
+
+MKL = 'nomkl'
+if [[ ! -z $(echo $CONDA_DEPENDENCIES | grep '\bmkl\b') ]]; then
+    MKL = ''
+fi
+
 if [[ $NUMPY_VERSION == dev* ]]; then
     # We install nomkl here to make sure that Numpy and Scipy versions
     # installed subsequently don't depend on the MKL. If we don't do this, then
     # we run into issues when we install the developer version of Numpy
     # because it is then not compiled against the MKL, and one runs into issues
     # if Scipy *is* still compiled against the MKL.
-    conda install $QUIET --no-pin $PYTHON_OPTION nomkl
+    conda install $QUIET --no-pin $PYTHON_OPTION $MKL
     # We then install Numpy itself at the bottom of this script
-    export CONDA_INSTALL="conda install $QUIET $PYTHON_OPTION"
+    export CONDA_INSTALL="conda install $QUIET $PYTHON_OPTION $MKL"
 elif [[ $NUMPY_VERSION == stable ]]; then
-    conda install $QUIET --no-pin numpy=$LATEST_NUMPY_STABLE
+    conda install $QUIET --no-pin numpy=$LATEST_NUMPY_STABLE $MKL
     export NUMPY_OPTION="numpy=$LATEST_NUMPY_STABLE"
-    export CONDA_INSTALL="conda install $QUIET $PYTHON_OPTION numpy=$LATEST_NUMPY_STABLE"
+    export CONDA_INSTALL="conda install $QUIET $PYTHON_OPTION numpy=$LATEST_NUMPY_STABLE $MKL"
 elif [[ $NUMPY_VERSION == pre* ]]; then
-    conda install $QUIET --no-pin nomkl numpy
+    conda install $QUIET --no-pin $MKL numpy
```

Comment:
So in the case of Numpy dev and prerelease, we actually want to always force nomkl right?

Reply:
👍